custom/ilsgateway/api.py

```diff
@@ -549,9 +549,14 @@ def get_or_create_msd_zone(region):
             location.save()
 
             interface = SupplyInterface(self.domain)
-            if ilsgateway_location.type == 'FACILITY' and not interface.get_by_location(location):
-                interface.create_from_location(self.domain, location)
-                location.save()
+            if ilsgateway_location.type == 'FACILITY':
+                if not interface.get_by_location(location):
+                    interface.create_from_location(self.domain, location)
+                    location.save()
+                else:
+                    sql_location = location.sql_location
+                    if not sql_location.supply_point_id:
+                        location.save()
```

Comment:
can you explain how this fixes it - is the supply point automatically created on save()?

also, if that's true, do we need to resync the locations in order to fix the ones with it missing?